### ISSUE TITLE
False Error with strum #[derive(Display)]: enum doesn't implement Display

### ISSUE THREAD

**Issue by pedrofgodinho** [intellij-rust/intellij-rust#8847]
## Environment

* **IntelliJ Rust plugin version:** 0.4.170.4627-221
* **Rust toolchain version:** 1.60.0
* **IDE name and version:** CLion 2022.1.1
* **Operating system:** Windows 11

## Problem description

After using strum to derive Display causes the IDE to report that enum does not implement the Display trait when trying to print it. 
![image](https://user-images.githubusercontent.com/5845396/168827777-e12ce159-362a-4183-8f2c-49c3a620f36d.png)

## Steps to reproduce

This code compiles and runs correctly, yet reports an error both with and without external linter enabled. 

```toml
[package]
name = "bugexample"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
strum = "0.24"
strum_macros = "0.24"
```

```rust
use strum_macros::Display;

#[derive(Display)]
enum Example {
    VariantA,
    VariantB
}

fn main() {
    println!("{} {}", Example::VariantA, Example::VariantB);
}
```

**Comment by Azgrom** [comment]
Hey @pedrofgodinho have you tried invalidating the cache as suggested by @Undin [here](https://github.com/intellij-rust/intellij-rust/issues/8845#issuecomment-1128455243)?

**Comment by pedrofgodinho** [comment]
I have. Invalidating cache does not seem to fix the issue, as the errors come back as soon as it finished indexing.

**Comment by Undin** [comment]
> Hey @pedrofgodinho have you tried invalidating the cache as suggested by @Undin here?

It shouldn't help in this case since the reasons differ. Cache invalidation may help in some weird situations when something stops working unexpectedly (for example, after upgrading). In some cases, it's caused by cache corruption and, as a result, cache invalidation helps

The current example is about procedural macro support. And the support is not enabled by default for now
@pedrofgodinho Try to enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-faq.html#experimental-features) features to turn it on and reload the project model via `Refresh Cargo Projects` action. See #6908 for more details

**Comment by pedrofgodinho** [comment]
Enabling the experimental features you mentioned fixed it! 

I'm closing this issue, thank you for your time @Undin

**Comment by Undin** [comment]
Let's keep it open since procedural macro expansion is not enabled by default yet
